src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/AssemblyPart.cs

```diff
@@ -42,6 +51,27 @@ public AssemblyPart(Assembly assembly)
         public IEnumerable<TypeInfo> Types => Assembly.DefinedTypes;
 
         /// <inheritdoc />
+        public IReadOnlyCollection<PrecompiledViewInfo> PrecompiledViews
+        {
+            get
+            {
+                var precompiledAssemblyName = new AssemblyName(Assembly.FullName);
+                precompiledAssemblyName.Name = precompiledAssemblyName.Name + PrecompiledViewsAssemblySuffix;
+
+                var viewFactoryTypeName = $"{ViewCollectionNamespace}.{ViewCollectionTypeName},{precompiledAssemblyName}";
+                var viewFactoryType = Type.GetType(viewFactoryTypeName);
+
+                if (viewFactoryType == null)
+                {
+                    return null;
+                }
+
+                var precompiledViews = Activator.CreateInstance(viewFactoryType) as PrecompiledViews;
```

Comment:
Seems like this should be a c-style cast. You want to get an exception if this fails.